third_party/winuwp_h264/H264Encoder/H264Encoder.cc

```diff
@@ -329,14 +324,33 @@ int WinUWPH264EncoderImpl::Encode(
   const VideoFrame& frame,
   const CodecSpecificInfo* codec_specific_info,
   const std::vector<FrameType>* frame_types) {
+  if (!inited_) {
+    return WEBRTC_VIDEO_CODEC_UNINITIALIZED;
+  }
+
   {
-      rtc::CritScope lock(&crit_);
-      if (!inited_) {
-      return -1;
+    rtc::scoped_refptr<webrtc::VideoFrameBuffer> frame_buffer =
+        frame.video_frame_buffer();
+    int cur_width = frame_buffer->width();
+    int cur_height = frame_buffer->height();
+    int64_t now = rtc::TimeMillis();
+
+    rtc::CritScope lock(&crit_);
+    // Reset the encoder configuration if necessary.
+    bool res_changed = cur_width != (int)width_ || cur_height != (int)height_;
+    bool should_change_rate_now =
+        rate_change_requested_ &&
+        (now - last_rate_change_time_rtc_ms) > kMinIntervalBetweenRateChangesMs;
```

Comment:
Should this be inside a `#ifdef DYNAMIC_BITRATE` for consistency?

Reply:
That define seems like a testing thing. I'd rather leave it in just one place.